cloudify_agent/app.py

```diff
@@ -81,17 +83,30 @@ def callback(*args, **kwargs):
     sender.hub.call_soon(callback=callback)
 
 
+class HeartbeatCelery(Celery):
+    """A Celery app that sends AMQP heartbeats."""
+
+    # celery has no public API to set the AMQP heartbeat
+    # Note: this is different than the --heartbeat-interval flag to the worker
+    # which sets celery-level interval, which is for keeping in sync with
+    # other workers, while the queue is up.
+    # This is AMQP-level interval, which is for detecting the queue being
+    # down.
+    def connection(self, *args, **kwargs):
+        kwargs['heartbeat'] = 5
+        return super(HeartbeatCelery, self).connection(*args, **kwargs)
+
+
 # This attribute is used as the celery App instance.
 # it is referenced in two ways:
 #   1. Celery command line --app options.
 #   2. cloudify.dispatch.dispatch uses it as the 'task' decorator.
 # For app configuration, see cloudify.broker_config.
-
-
-app = Celery()
+app = HeartbeatCelery()
```

Comment:
doesn't hurt to have it even in the no-cluster case - we'll just gain earlier detection of unclean disconnects